evaluator/processor.py

```diff
@@ -56,11 +64,62 @@ def __init__(
         self.remediations_results = remediations_results
         self.evaluator_results = evaluator_results
         self.payload_tracker = payload_tracker
+        self.recent_recalc_events: Dict[int, RecalcEvent] = {}
 
     async def init(self):
         """Async constructor"""
         await self.evaluator_logic.init()
 
+    async def _get_changed_packages(self, recalc_event_id: Optional[int], conn: AsyncConnection) -> Optional[dict]:
+        """Get changed packages for a recalc event from cache or fetch it from DB."""
+        if recalc_event_id is None:
+            return None
+        if recalc_event_id not in self.recent_recalc_events:
+            # Cleanup old recalc events first
+            for rc_e_id in list(self.recent_recalc_events):
+                created = self.recent_recalc_events[rc_e_id].created
+                if created < datetime.now() - timedelta(days=30):
```

Comment:
nitpick: `datetime.now() - timedelta(days=30)` can be calculated before the loop

Reply:
fixed